### ISSUE TITLE
[feature request] warnings for functions with unspecified dim arguments

### ISSUE THREAD

**Issue by christinaheinze** [pytorch/pytorch#5266]
Hi, 

I noticed that when calling `torch.sort()` without specifying the `dim` argument results in undesired behavior when called on a tensor of dimension nx1. If `dim` is not given in `torch.sort()`, the last dimension of the input is chosen. This is documented but not desirable when the last dimension is 1. A short example for this is the following: 

```
import torch
a = torch.randn(10, 1)
print(a)
a_s, _ = torch.sort(a)
print(a_s)
```
yields the same (unsorted) tensor for both `print(a)` and `print(a_s)`
```
 1.2201
 0.0733
 1.6214
 0.6704
 0.8339
-1.1773
-0.4946
 0.4496
 0.7240
 1.2748
[torch.FloatTensor of size 10x1]


 1.2201
 0.0733
 1.6214
 0.6704
 0.8339
-1.1773
-0.4946
 0.4496
 0.7240
 1.2748
[torch.FloatTensor of size 10x1]
```

We (that is me, @Kaixhin and @bastings) would suggest to issue a warning in such a case to improve usability. 

Cheers, 
Christina

**Comment by bastings** [comment]
+1
Maybe we should generally warn when a function has dim as an argument and it is not explicitly set (as is the case with softmax, too).
